fix: Release Notes Generator fails with missing @anthropic-ai/sdk

## Problem

The \`Release Notes Generator\` workflow fails on tag push with:

\`\`\`
Cannot find module '@anthropic-ai/sdk' from '.github/scripts/generate-release-notes.ts'
\`\`\`

**Failed run**: https://github.com/baekenough/oh-my-customcode/actions/runs/21743535623

## Root Cause

The \`generate-release-notes.ts\` script imports \`@anthropic-ai/sdk\`, but the workflow's \`bun install\` step likely doesn't install it (may be using \`--production\` flag, or the dependency is missing from \`package.json\`).

This is the same class of issue that was fixed for \`docs-validator.yml\` in PR #33 (removed \`--production\` flag).

## Fix

1. Check if \`@anthropic-ai/sdk\` is in \`dependencies\` or \`devDependencies\` in \`package.json\`
2. Ensure the workflow install step uses \`bun install\` (not \`bun install --production\`)
3. Verify the \`RELEASE_BOT\` secret provides the correct Anthropic API key

## Related

- PR #33: Fixed same issue for docs-validator (\`bun install --production\` → \`bun install\`)
- Workflow: \`.github/workflows/release-notes-generator.yml\`
- Script: \`.github/scripts/generate-release-notes.ts\`

--- a/.github/workflows/release-notes.yml
+++ b/.github/workflows/release-notes.yml
@@ -32,7 +32,7 @@ jobs:
         uses: oven-sh/setup-bun@v2
 
       - name: Install dependencies
-        run: bun install --production
+        run: bun install
 
       - name: Extract version from tag
         id: version

--- a/.github/workflows/reusable-issue-analyzer.yml
+++ b/.github/workflows/reusable-issue-analyzer.yml
@@ -70,7 +70,7 @@ jobs:
         uses: oven-sh/setup-bun@v2
 
       - name: Install dependencies
-        run: bun install --production
+        run: bun install
 
       - name: Analyze issue
         env:

